app/upgrades/v8/upgrades.go

```diff
@@ -16,6 +18,15 @@ func CreateUpgradeHandler(
 	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 		ctx.Logger().Info("start to run module migrations...")
 
+		// Add atom name and symbol into the bank keeper
+		atomMetaData, found := keepers.BankKeeper.GetDenomMetaData(ctx, "uatom")
+		if !found {
+			return nil, errors.New("atom denom not found")
+		}
+		atomMetaData.Name = "Cosmos Hub Atom"
+		atomMetaData.Symbol = "ATOM"
```

Comment:
Hi, when i query the node in public net, denom-metadata is already there.
```
{"metadatas":[{"description":"The native staking token of the Cosmos Hub.","denom_units":[{"denom":"uatom","exponent":0,"aliases":["microatom"]},{"denom":"matom","exponent":3,"aliases":["milliatom"]},{"denom":"atom","exponent":6,"aliases":[]}],"base":"uatom","display":"atom","name":"","symbol":""}],"pagination":{"next_key":null,"total":"1"}}
```

also, the denom is uatom.

Reply:
I think this is different than the actual denom

Reply:
https://github.com/cosmos/cosmos-sdk/blob/55054282d2df794d9a5fe2599ea25473379ebc3d/x/bank/types/metadata_test.go

Reply:
We can confirm the parameter through mainnet API (`http://<rpc...>:1317/cosmos/bank/v1beta1/denoms_metadata`) . The parameter the SDK use as a key index is the base, and we are using `uatom`:


```shell
$ curl http://<RPC_ADDRESS>:1317/cosmos/bank/v1beta1/denoms_metadata
{
  "metadatas": [
    {
      "description": "The native staking token of the Cosmos Hub.",
      "denom_units": [
        {
          "denom": "uatom",
          "exponent": 0,
          "aliases": [
            "microatom"
          ]
        },
        {
          "denom": "matom",
          "exponent": 3,
          "aliases": [
            "milliatom"
          ]
        },
        {
          "denom": "atom",
          "exponent": 6,
          "aliases": [
          ]
        }
      ],
      "base": "uatom",
      "display": "atom",
      "name": "",
      "symbol": ""
    }
  ],
  "pagination": {
    "next_key": null,
    "total": "1"
  }
}
```

Reply:
https://github.com/cosmos/cosmos-sdk/blob/a975f67b245c214fa11408c4eb361147b4679d62/x/bank/keeper/keeper.go#L309-L309